### ISSUE TITLE
Change outputChannelCount dynamically after instantiation (and independently from input channelCount)

### ISSUE THREAD

**Issue by JohnWeisz** [WebAudio/web-audio-api#2438]
## Describe the feature

Allow changing [outputChannelCount](https://www.w3.org/TR/webaudio/#dom-audioworkletnodeoptions-outputchannelcount) dynamically, after and separately from instantiation.

The following example briefly describes a node which always takes a single stereo input, has a single stereo output (by default), and can switch between stereo output, quad output, and auto output on demand, dynamically, at any time:

```
class CustomChannelMixerNode extends AudioWorkletNode {
    constructor() {
        super("custom-channel-mixer-processor", {
            numberOfInputs: 1,
            numberOfOutputs: 1,
            channelCountMode: "explicit",
            channelCount: 2,
            outputChannelCount: [2]
        });
    }

    useStereoOutput() {
        if (this.outputChannelCount && this.outputChannelCount[0] === 2) {
            console.warn("already using stereo output.");
            return;
        }

        this.outputChannelCount = [2];
    }

    useQuadOutput() {
        if (this.outputChannelCount && this.outputChannelCount[0] === 4) {
            console.warn("already using quad output.");
            return;
        }

        this.outputChannelCount = [4];
    }

    useAutoOutput() {
        if (!this.outputChannelCount) {
            console.warn("already using auto output.");
            return;
        }

        this.outputChannelCount = null;
    }

    // ...
}
```

## Is there a prototype?

Can be done in a limited fashion using a wrapper object (and augmenting the `AudioNode.connect` method to accept the wrapper object), as currently a reinstatiation of the AudioWorkletNode is required to change output channel count.

For 1-1 input/output count, I strongly believe this idea can also be done by setting `channelCount` to the 
max of input/output channel count, and adding JS-implemented channel upmixing/downmixing before `process` calls.

## Describe the feature in more detail

_Note: below each reference to `this` refers to an AudioWorkletNode instance._

### General behavior

Assigning to `this.outputChannelCount` will result in the exact same behavior as if the AudioWorkletNode was initialized with the specified output channel configuration in the first place, beginning with the "next" `process` call on the corresponding AudioWorkletProcessor instance.

_Here, the "next" process call can be interpreted essentially the same as the next process call after setting `this.channelCount`._

### Immutability

Assigning to `this.outputChannelCount` is consistent with assigning to `this.channelCount`, although it is up to question how exactly assigning to a single specific output is handled, i.e. what the following does:

```
this.outputChannelCount[0] = 4;
```

My vote here is to freeze `this.outputChannelCount`, and always require a full array assignment.

### Clearing an explicitly defined value

In a case where the developer wants to clear any values set previously, and instead use an automatically determined outputChannelCount, `this.outputChannelCount` can be assigned `null`. _Personally, I feel this part of the proposal needs some more discussion._

This also implies the default value of `this.outputChannelCount` is `null` if not set to a specific configuration in the constructor.

**Comment by JohnWeisz** [comment]
> Would it be ok if the channel count change happened via messages from the main thread?

Yeah it would probably work fine. This is already possible to do BTW using a worklet processor that acts as a channel detector, reinstantiating and reconnecting the audio worklet that does the actual processing... or something similar along these lines.

**Comment by martinguder** [comment]
This bug seem to be resolved, except for some edge cases. Look here:
https://bugs.chromium.org/p/chromium/issues/detail?id=1428291#makechanges

**Comment by hoch** [comment]
The workaround suggested by https://github.com/WebAudio/web-audio-api/issues/2438#issuecomment-837375979 and https://github.com/WebAudio/web-audio-api/issues/2438#issuecomment-1120217706 does seem reasonable.
